sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/converter/AzureAmqpRetryOptionsConverter.java

```diff
@@ -15,27 +17,50 @@
 public final class AzureAmqpRetryOptionsConverter implements Converter<RetryOptionsProvider.AmqpRetryOptions, AmqpRetryOptions> {
 
     public static final AzureAmqpRetryOptionsConverter AMQP_RETRY_CONVERTER = new AzureAmqpRetryOptionsConverter();
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureAmqpRetryOptionsConverter.class);
+
 
     private AzureAmqpRetryOptionsConverter() {
 
     }
 
     @Override
     public AmqpRetryOptions convert(RetryOptionsProvider.AmqpRetryOptions retry) {
-        AmqpRetryOptions result = new AmqpRetryOptions();
+        PropertyMapper mapper = new PropertyMapper();
 
-        if (RetryOptionsProvider.RetryMode.EXPONENTIAL.equals(retry.getMode())) {
-            result.setMode(AmqpRetryMode.EXPONENTIAL);
-        } else if (RetryOptionsProvider.RetryMode.FIXED.equals(retry.getMode())) {
-            result.setMode(AmqpRetryMode.FIXED);
-        }
+        RetryOptionsProvider.RetryMode retryMode = retry.getMode();
 
-        PropertyMapper mapper = new PropertyMapper();
-        mapper.from(retry.getMaxRetries()).to(result::setMaxRetries);
-        mapper.from(retry.getTryTimeout()).to(result::setTryTimeout);
-        mapper.from(retry.getBaseDelay()).to(result::setDelay);
-        mapper.from(retry.getMaxDelay()).to(result::setMaxDelay);
+        if (RetryOptionsProvider.RetryMode.EXPONENTIAL.equals(retryMode)) {
+            RetryOptionsProvider.RetryOptions.ExponentialRetryOptions exponential = retry.getExponential();
+            if (exponential != null && exponential.getMaxRetries() != null) {
+                AmqpRetryOptions result = new AmqpRetryOptions();
+
+                result.setMode(AmqpRetryMode.EXPONENTIAL);
 
-        return result;
+                mapper.from(retry.getTryTimeout()).to(result::setTryTimeout);
```

Comment:
The AmqpRetryOptions has the default value for properties, if not applied, it will use the default value, should we add logs to tell the user which value is used?  Or ensure consistent configuration and behavior, we can add a default value for the `to` method, 
`mapper.from(retry.getTryTimeout()).to(result::setTryTimeout, null);`